liferay-frontend/liferay-portal
# LPS-161236 Adds missing closeOnClickOutside functionallity

> Depends on https://github.com/liferay-frontend/liferay-portal/pull/2676/commits. I made a lot of modifications, and I don't want to rebase again

# What It does:

I was relying on ClayPopover's closeOnClickOutside API brereading the docsain, I noticed:

> [Flag to indicate if the popover should be closed when clicking outside; only works if used with trigger](https://clayui.com/docs/components/popover/api.html)

So, I was not using trigger property due to some limitations(a.k.a positioning the Popover outside React tr,ee) and I needed to implemeit nt on my own

# How to test my changes

1. Go to Global Menu -> Site -> "Go Site Settings" of Liferay DXP site
2. Click on Walkthrough
3. Check "Enable Walkthrough"
4. Paste the following JSON snippet on text area: 

```json
{
        closeOnClickOutside: true,
	pages: {
		'/home' : ['step-1', 'step-2'],
	},
	steps: [
		{
			content: '<span>Content 1</span><br/><code>Hello1</code>',
			darkbg: false,
			id: 'step-1',
			nodeToHighlight: '.logo',
			title: 'Title 1',
		},
		{
			content: '<span>Content 2</span><br/><code>Hello2</code>',
			darkbg: false,
			id: 'step-2',
			nodeToHighlight: '[aria-label="Open menu"]',
			positioning: 'left',
			title: 'Title 2',
		}
	],
}
```

5. Click on "Update" and go back Home page
6. Click on Hotspot and then Next
7. Click outside the popover area, and it will close :)
